.github/workflows/ci.yml

```diff
@@ -907,3 +912,275 @@ jobs:
+        if [[ "${cling_on}" == "ON" ]]; then
+          cd build
+          cmake -DLLVM_ENABLE_PROJECTS="clang;lld"               \
+                -DLLVM_EXTERNAL_PROJECTS=cling                     \
+                -DLLVM_EXTERNAL_CLING_SOURCE_DIR=../../cling       \
+                -DLLVM_TARGETS_TO_BUILD="WebAssembly;host;NVPTX"   \
+                -DCMAKE_BUILD_TYPE=Release                         \
+                -DLLVM_ENABLE_ASSERTIONS=ON                        \
+                -DCLANG_ENABLE_STATIC_ANALYZER=OFF                 \
+                -DCLANG_ENABLE_ARCMT=OFF                           \
+                -DCLANG_ENABLE_FORMAT=OFF                          \
+                -DCLANG_ENABLE_BOOTSTRAP=OFF                       \
+                -DLLVM_ENABLE_ZSTD=OFF                             \
+                -DLLVM_ENABLE_TERMINFO=OFF                         \
+                -DLLVM_ENABLE_LIBXML2=OFF                          \
+                ../llvm
+          cmake --build . --target lld --parallel ${{ env.ncpus }}
+          cmake --build . --target clang --parallel ${{ env.ncpus }}
+          cmake --build . --target cling --parallel ${{ env.ncpus }}
+          # Now build gtest.a and gtest_main for CppInterOp to run its tests.
+          cmake --build . --target gtest_main --parallel ${{ env.ncpus }}
+        else
+          # Apply patches
+          git apply -v ../patches/llvm/clang${{ matrix.clang-runtime }}-*.patch
+          echo "Apply clang${{ matrix.clang-runtime }}-*.patch patches:"
+          cd build
+          cmake -DLLVM_ENABLE_PROJECTS="clang;lld"                  \
+                -DLLVM_TARGETS_TO_BUILD="WebAssembly;host;NVPTX"    \
+                -DCMAKE_BUILD_TYPE=Release                          \
+                -DLLVM_ENABLE_ASSERTIONS=ON                         \
+                -DCLANG_ENABLE_STATIC_ANALYZER=OFF                  \
+                -DCLANG_ENABLE_ARCMT=OFF                            \
+                -DCLANG_ENABLE_FORMAT=OFF                           \
+                -DCLANG_ENABLE_BOOTSTRAP=OFF                        \
+                -DLLVM_ENABLE_ZSTD=OFF                              \
+                -DLLVM_ENABLE_TERMINFO=OFF                          \
+                -DLLVM_ENABLE_LIBXML2=OFF                           \
+                ../llvm
+          cmake --build . --target all --parallel ${{ env.ncpus }}
+        fi
+        cd ../../
+
+    - name: Save Cache LLVM/Clang runtime build directory
+      uses: actions/cache/save@v3
+      if: ${{ steps.cache.outputs.cache-hit != 'true' }}
+      with:
+        path: |
+          llvm-project
+          ${{ matrix.cling=='On' && 'cling' || '' }}
+        key: ${{ steps.cache.outputs.cache-primary-key }}
+        
+    - name: Emscripten build of CppInterOp on Unix systems
+      if: ${{ runner.os != 'windows' }}
+      shell: bash -l {0}
+      run: |
+        emsdk activate latest
```

Comment:
Let's use emscripten `3.1.45` ? 
I guess we should only use `3.1.45` ([reference](https://github.com/search?q=repo%3Aemscripten-forge%2Frecipes+3.1.45&type=code))
(ie the same one as in `emscripten-forge`) . All repos like [xeus-cpp](https://github.com/compiler-research/xeus-cpp/blob/216dbed46e358d82fd8fe3065b4533f3e1c49119/.github/workflows/main.yml#L156), [xeus-python](https://github.com/jupyter-xeus/xeus-python/blob/1c55a64050ad43e02611e0a77f2bd1a2a6a00bf5/.github/workflows/main.yml#L167) etc follow that as we have dependencies based on emscripten-forge.

Reply:
I will make this change sometime today.